src/snowflake/snowpark/modin/plugin/_internal/resample_utils.py

```diff
@@ -667,20 +671,29 @@ def perform_asof_join_on_frame(
     # are the steps to take:
 
     # 1. Construct right_frame using referenced_frame, which has a
-    # temporary column, interval_end_col, that olds the closest
+    # temporary column, interval_end_col, that holds the closest
     # following timestamp to every value in __index__. The last value in
-    # interval_end_col is dummy value that represents the largest
-    # possible date in Snowflake.
+    # interval_end_col is dummy value that represents the smallest or largest
+    # (e.g. bfill or ffill) possible date in Snowflake.
     interval_end_pandas_label = "interval_end_col"
     interval_start_snowflake_quoted_identifier = (
         get_snowflake_quoted_identifier_for_resample_index_col(referenced_frame)
     )
-    interval_end_col = coalesce(
-        lead(col(interval_start_snowflake_quoted_identifier)).over(
-            Window.order_by(col(interval_start_snowflake_quoted_identifier).asc())
-        ),
-        pandas_lit("9999-01-01 00:00:00"),
-    )
+    if fill_method == "bfill":
+        interval_end_col = coalesce(
+            lag(col(interval_start_snowflake_quoted_identifier)).over(
+                Window.order_by(col(interval_start_snowflake_quoted_identifier).asc())
+            ),
+            pandas_lit("1582-01-01 00:00:00"),
+        )
+    else:
+        assert fill_method == "ffill"
```

Comment:
I don't think it's really necessary since it's functionality that is supported as part of `fillna` and I don't really see any downside, but I can add it if you think it's better?